### ISSUE TITLE
false negative Lint/DuplicateMethods in blocks

### ISSUE THREAD

**Issue by TSMMark** [rubocop/rubocop#11829]
Methods defined in a block are not identified. 

```ruby
def included(includer)
  includer.include(SomeModule)
end

def included(includer) # <-- Reported
  includer.include(SomeModule)
end

module SomeClass

  def self.included(includer)
    includer.include(SomeModule)
  end

  def self.included(includer) # <-- Reported
    includer.include(SomeModule)
  end

end

SomeSharedExample = Minitest::SharedExamples.new do
  def self.included(includer)
    includer.include(SomeModule)
  end

  def self.included(includer) # <-- Not reported
    includer.include(SomeModule)
  end

  def foobar(includer)
    includer.include(SomeModule)
  end

  def foobar(includer) # <-- Not reported
    includer.include(SomeModule)
  end
end
```

--------

## Expected behavior

Should register offense when duplicate methods defined within a block.

## Actual behavior

False negative

## Steps to reproduce the problem

See code example above

## RuboCop version

1.50.2

**Comment by alexeyschepin** [comment]
The bug can be replicated by the following test in `spec/rubocop/cop/lint/duplicate_methods_spec.rb`
```ruby
include_examples('in scope', 'dynamic class with module', 'A = Module::Class.new do')
```

`Module::Class.new` is recognised as `dsl` by `possible_dsl?` method in `lib/rubocop/cop/lint/duplicate_methods.rb`.
`ancestor.class_constructor?` is not `true` as it would be for `Class.new`.

@koic can you point me what would be the right solution here? I assume you want to keep `class_constructor?` method as it is.
